addons/hr_holidays/static/src/persona_model_patch.js

```diff
@@ -1,10 +1,43 @@
 import { Persona } from "@mail/core/common/persona_model";
-import { deserializeDateTime } from "@web/core/l10n/dates";
+import { deserializeDate, deserializeDateTime } from "@web/core/l10n/dates";
 import { _t } from "@web/core/l10n/translation";
 import { patch } from "@web/core/utils/patch";
 
 const { DateTime } = luxon;
 
+export function getOutOfOfficeDateEndText(date) {
+    const { days, weeks } = deserializeDate(date)
+          .diff(DateTime.now().startOf("day"), ["days", "weeks"])
+          .toObject();
+    if (weeks) {
+        if (weeks === 1 && !days) {
+            return _t("Back in 1 week");
+        }
+        if (weeks > 1 && !days) {
+            return _t("Back in %(weeks)s weeks", { weeks });
+        }
+        if (days === 1 && weeks === 1) {
+            return _t("Back in 1 week and 1 day");
+        }
+        if (days > 1 && weeks > 1) {
+            return _t("Back in %(weeks)s weeks and %(days)s days", { weeks, days });
+        }
+        if (weeks > 1 && days === 1) {
+            return _t("Back in %(weeks)s weeks and 1 day", { weeks });
+        }
+        if (weeks === 1 && days > 1) {
+            return _t("Back in 1 week and %(days)s days", { days });
+        }
+    }
+    if (days === 1) {
+        return _t("Back in 1 day");
+    }
+    if (days > 1) {
```

Comment:
(nitpicking) I know that it's equivalent, but I'd use `else if` instead of just `if` to make things more explicit/readable